src/BladeDirectivesServiceProvider.php

```diff
@@ -32,6 +28,14 @@ public function register()
             return "<?php try { echo \$__env->first({$expression}, \Illuminate\Support\Arr::except(get_defined_vars(), ['__data', '__path']))->render(); } catch (\InvalidArgumentException \$e) { if (!app()->environment('production')) { echo '<hr>'.__('View not found: :view', ['view' => implode(', ', [{$expression}][0])]).'<hr>'; } } ?>";
         });
 
+        Blade::directive('markdown', function ($markdown) {
+            return "<?php echo (fn(\$markdown, \$html_input = 'escape') => Str::markdown(\$markdown ?? '', ['html_input' => \$html_input]))($expression); ?>";
+        });
+
+        Blade::directive('return', function () {
+            return "<?php return; ?>";
+        });
```

Comment:
I feel like you have already tried this, but wouldn't it already work to do
```suggestion
        Blade::directive('markdown', function ($markdown, $html_input = 'escape') {
            return "<?php echo Str::markdown($markdown ?? '', ['html_input' => $html_input]); ?>";
        });

        Blade::directive('return', function () {
            return "<?php return; ?>";
        });
```

Reply:
Yeah this does not work, it generated the following code:

```blade
@markdown($footer->text, 'strip')
```

```php
<?php echo Str::markdown($footer->text, 'strip' ?? '', ['html_input' => 'escape']); ?>
```

Reply:
Using something like `explode()` is also a bad idea obviously, as any comma in the the markdown text would break it.

Reply:
Yeah i agree, using explode would be a bad idea especially as it's good and it works as-is